pkg/descheduler/descheduler.go

```diff
@@ -295,6 +287,7 @@ func RunDeschedulerLoop(ctx context.Context, rs *options.DeschedulerServer, node
 func RunProfiles(ctx context.Context, nodes []*v1.Node, deschedulerPolicy *api.DeschedulerPolicy, client clientset.Interface,
 	cycleSharedInformerFactory informers.SharedInformerFactory, podEvictor *evictions.PodEvictor, getPodsAssignedToNode podutil.GetPodsAssignedToNodeFunc,
 ) {
+	frameworkprofiles := []*frameworkprofile.ProfileImpl{}
```

Comment:
```go
type eprunner func(ctx context.Context, nodes []*v1.Node) *frameworktypes.Status
var descheduleEPs, balanceEPs []eprunner
for _, profile := range deschedulerPolicy.Profiles {
	currProfile, err := frameworkprofile.NewProfile(...)
	...
	descheduleEPs = append(descheduleEPs, currProfile.RunDeschedulePlugins)
	balanceEPs = append(balanceEPs, currProfile.RunBalancePlugins)
}
for _, epf := range descheduleEPs {
	status := epf(ctx, nodes)
	...
}
for _, epf := range balanceEPs {
	status := epf(ctx, nodes)
	...
}
``` 
or similar to avoid to make the `profileImpl` exported.

Reply:
Do we want to print the profile name in the error? if yes this can get O(n²) for each extension point....

Like

```Go
	type eprunner func(ctx context.Context, nodes []*v1.Node) *frameworktypes.Status
	var descheduleEPs, balanceEPs []eprunner
	deschByProfile := make(map[string][]eprunner)
	balanceByProfile := make(map[string][]eprunner)
	for _, profile := range deschedulerPolicy.Profiles {
		currProfile, err := frameworkprofile.NewProfile(
			profile,
			pluginregistry.PluginRegistry,
			frameworkprofile.WithClientSet(client),
			frameworkprofile.WithSharedInformerFactory(cycleSharedInformerFactory),
			frameworkprofile.WithPodEvictor(podEvictor),
			frameworkprofile.WithGetPodsAssignedToNodeFnc(getPodsAssignedToNode),
		)
		if err != nil {
			klog.ErrorS(err, "unable to create a profile", "profile", profile.Name)
			continue
		}
		descheduleEPs = append(descheduleEPs, currProfile.RunDeschedulePlugins)
		deschByProfile[profile.Name] = descheduleEPs
		balanceEPs = append(balanceEPs, currProfile.RunBalancePlugins)
		balanceByProfile[profile.Name] = balanceEPs

	}

	for profileName, eprunners := range deschByProfile {
		// First deschedule
		for _, epf := range eprunners {
			status := epf(ctx, nodes)
			if status != nil && status.Err != nil {
				klog.ErrorS(status.Err, "running deschedule extension point failed with error", "profile", profileName)
				continue
			}
		}
	}
```

Reply:
You can always create additional indexed list with the profile names.

Reply:
still O(n²) loops for each new EP we will add in the future

Reply:
sorry misread RunDeschedulePlugins